diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/EmptyBlock.kt

```diff
@@ -0,0 +1,103 @@
+import com.pinterest.ktlint.core.ast.ElementType.FOR
+import com.pinterest.ktlint.core.ast.ElementType.LBRACE
+import com.pinterest.ktlint.core.ast.ElementType.MODIFIER_LIST
+import com.pinterest.ktlint.core.ast.ElementType.OBJECT_DECLARATION
+import com.pinterest.ktlint.core.ast.ElementType.OVERRIDE_KEYWORD
+import com.pinterest.ktlint.core.ast.ElementType.RBRACE
+import com.pinterest.ktlint.core.ast.ElementType.SECONDARY_CONSTRUCTOR
+import com.pinterest.ktlint.core.ast.ElementType.WHEN
+import com.pinterest.ktlint.core.ast.ElementType.WHILE
+import com.pinterest.ktlint.core.ast.ElementType.WHITE_SPACE
+import org.cqfn.diktat.common.config.rules.RuleConfiguration
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.common.config.rules.getRuleConfig
+import org.cqfn.diktat.ruleset.constants.Warnings.BRACES_BLOCK_STRUCTURE_ERROR
+import org.cqfn.diktat.ruleset.utils.hasChildOfType
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.LeafPsiElement
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.PsiWhiteSpaceImpl
+
+class EmptyBlock : Rule("block-structure") {
+
+    companion object {
+        private const val emptyBlockSize = 4
+    }
+
+    private lateinit var configRules: List<RulesConfig>
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
+    private var fileName: String? = null
+    private var isFixMode: Boolean = false
+
+    override fun visit(node: ASTNode,
+                       autoCorrect: Boolean,
+                       params: KtLint.Params,
+                       emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+        configRules = params.getDiktatConfigRules()
+        fileName = params.fileName
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        val configuration = EmptyBlockStyleConfiguration(
+                configRules.getRuleConfig(BRACES_BLOCK_STRUCTURE_ERROR)?.configuration ?: mapOf()
+        )
+        searchNode(node, configuration)
+    }
+
+    private fun searchNode(node: ASTNode, configuration: EmptyBlockStyleConfiguration) {
+        val newNode = when (node.elementType) {
+            THEN -> node.findChildByType(BLOCK)?.findChildByType(LBRACE)!!.treeNext.treeParent
+            ELSE -> node.findChildByType(BLOCK)?.findChildByType(LBRACE)!!.treeNext.treeParent
+            WHEN -> node.findChildByType(LBRACE)!!.treeNext.treeParent
+            FOR, WHILE, DO_WHILE -> node.findChildByType(BODY)?.findChildByType(BLOCK)?.findChildByType(LBRACE)!!.treeNext.treeParent
+            CLASS, OBJECT_DECLARATION -> node.findChildByType(CLASS_BODY)!!.findChildByType(LBRACE)!!.treeNext.treeParent
+            else -> if (node.hasChildOfType(BLOCK))
+                node.findChildByType(BLOCK)?.findChildByType(LBRACE)!!.treeNext.treeParent
+            else
+                null
```

Comment:
```suggestion
            else -> if (node.hasChildOfType(BLOCK)) node.findChildByType(BLOCK)?.findChildByType(LBRACE)!!.treeNext.treeParent else null
```
Please remember our own code style: either surround if-else by braces or use it ternary-style.